cocos/core/data/object.ts

```diff
@@ -318,8 +318,8 @@ class CCObject implements EditorExtendableObject {
         }
 
         if (JSB) {
-            // @ts-expect-error JSB method
-            this._destroy();
+            // TODO: `_destroy` method only implemented on native @dumganhar
+            (this as any)._destroy();
```

Comment:
please take a look at these TODO with `@dumganhar`

@dumganhar

Reply:
Yes, it's a method only used for native. Therefore, it's inside the `JSB` scope.